sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Queues/tests/QueueListenerTests.cs

```diff
@@ -648,6 +649,175 @@ public async Task ProcessMessageAsync_FunctionInvocationFails()
             await _listener.ProcessMessageAsync(_queueMessage, TimeSpan.FromMinutes(10), cancellationToken);
         }
 
+        private (QueueListener Listener,
+            TestQueueProcessor Processor,
+            CancellationTokenSource CallerCts,
+            CancellationTokenSource SystemShutdownCts)
+        CreateListenerAndMocks()
+        {
+            var callerCts = new CancellationTokenSource();
+            var systemShutdownCts = new CancellationTokenSource();
+            QueueProcessorOptions options = new QueueProcessorOptions(_mockQueue.Object, _loggerFactory, _queuesOptions);
+            var queueProcessor = new TestQueueProcessor(options);
+
+            _mockTriggerExecutor.Setup(x => x.ExecuteAsync(It.IsAny<QueueMessage>(), It.IsAny<CancellationToken>()))
+                               .ReturnsAsync(new FunctionResult(true));
+
+            var exceptionHandlerMock = new Mock<IWebJobsExceptionHandler>();
+
+            var listener = new QueueListener(
+               _mockQueue.Object,
+               null,
+               _mockTriggerExecutor.Object,
+               _mockExceptionDispatcher.Object,
+               _loggerFactory,
+               null,
+               _queuesOptions,
+               queueProcessor,
+               new FunctionDescriptor { Id = TestFunctionId },
+               null,
+               drainModeManager: null,
+               shutdownCancellationTokenSource: systemShutdownCts);
+
+            return (listener, queueProcessor, callerCts, systemShutdownCts);
+        }
+
+        [Test]
+        public async Task ProcessMessageAsync_NoTokensCanceled_CallsCompleteProcessingMessageAsync()
+        {
+            // Arrange
+            var (listener, processor, callerCts, shutdownCts) = CreateListenerAndMocks();
+
+            // Act
+            // Neither token is canceled
+            await listener.ProcessMessageAsync(_queueMessage, TimeSpan.FromMinutes(2), callerCts.Token);
+
+            // TODO verify that the message was processed and deleted
```

Comment:
Will this be done in this PR?

Reply:
Yes I'm trying to see if I can test this behavior without adding the cancellationTokenSources to the constructor :)